### PR TITLE
Using Structured Documentation for Signature Help and improving Parameter Documentation

### PR DESCRIPTION
**Pull request by akshita31**
Fixes : #1940 

Using the structured documentation to display parsed xml in signature help. Also displaying the parameter documentation properly.

 Omnisharp-roslyn side : https://github.com/OmniSharp/omnisharp-roslyn/pull/1085

### CODE REVIEW

**Comment by akshita31** [comment]
With this change, the signature help now appears like 
sig_help1
sig_help2

File: src/features/signatureHelpProvider.ts [hunk 1]
```diff
@@ -44,4 +45,14 @@ export default class OmniSharpSignatureHelpProvider extends AbstractSupport impl
             return ret;
         });
     }
+
+    private GetParameterDocumentation(parameter: SignatureHelpParameter) {
+        let summary = parameter.Documentation;
+        if (summary.length > 0) {
+            let nameInBold = "**" + parameter.Name + "**" + ": ";
```
Comment (rchande) [inline src/features/signatureHelpProvider.ts]: Use interpolated string?

File: src/omnisharp/protocol.ts [hunk 2]
```diff
@@ -214,6 +214,22 @@ export interface FindSymbolsResponse {
...
 }
 
+export interface DocumentationItem {
+    Name: string;
+    Documentation: string;
+}
+
+export interface DocumentationComment {
+    SummaryText: string ;
+    TypeParamElements :DocumentationItem[];
```
Comment (rchande) [inline src/omnisharp/protocol.ts]: Spacing

File: src/omnisharp/protocol.ts [hunk 3]
```diff
@@ -214,6 +214,22 @@ export interface FindSymbolsResponse {
     QuickFixes: SymbolLocation[];
 }
 
+export interface DocumentationItem {
+    Name: string;
+    Documentation: string;
+}
+
+export interface DocumentationComment {
+    SummaryText: string ;
```
Comment (rchande) [inline src/omnisharp/protocol.ts]: Spacing

File: src/features/signatureHelpProvider.ts [hunk 4]
```diff
@@ -49,7 +49,7 @@ export default class OmniSharpSignatureHelpProvider extends AbstractSupport impl
     private GetParameterDocumentation(parameter: SignatureHelpParameter) {
         let summary = parameter.Documentation;
         if (summary.length > 0) {
-            let nameInBold = "**" + parameter.Name + "**" + ": ";
+            let nameInBold = `**${parameter.Name}**: `;
```
Comment (rchande) [inline src/features/signatureHelpProvider.ts]: You can move the summary in here too.

**Comment by akshita31** [comment]
Yes right.

File: test/integrationTests/signatureHelp.integration.test.ts [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+        let fileName = 'sigHelp.cs';
+        let dir = path.dirname(testAssetWorkspace.projects[0].projectDirectoryPath);
+        let loc = path.join(dir, fileName);
+        fileUri = vscode.Uri.file(loc);
+        await omnisharp.waitForEmptyEventQueue();
+        await vscode.commands.executeCommand("vscode.open", fileUri);
+    });
+
+    
+    test("Returns response with documentation as undefined when method does not have documentation", async function () {
```
Comment (rchande) [inline test/integrationTests/signatureHelp.integration.test.ts]: There are two types of tests we should be thinking about:
* Unit tests, which test the specifics of the C# extension's signature help implementation. These tests should theoretically be able to run outside of VS Code and shouldn't need to send VS Code commands
* Integration tests, which verify that the C# extension's signature help is hooked up in VS Code and validate any nuances there (eg, trigger characters, etc). By definition, these tests run in VS Code.

Today, we can only test features by writing "integration" tests. At some point, we should think about what we'll need to do to write "unit" tests to test these kinds of implementation details. Not blocking your PR on this, but something to consider (cc @TheRealPiotrP ).

File: test/integrationTests/signatureHelp.integration.test.ts [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+
+    test("Returns documentation for the parameters", async function () {
+        let c = <vscode.SignatureHelp>await vscode.commands.executeCommand("vscode.executeSignatureHelpProvider", fileUri, new vscode.Position(19, 19));
+        let param1 = `**gameObject**: The game object.`;
+        let param2 = `**tagName**: Name of the tag.`;
+        expect((<vscode.MarkdownString> c.signatures[0].parameters[0].documentation).value).to.equal(param1);
+        expect((<vscode.MarkdownString> c.signatures[0].parameters[1].documentation).value).to.equal(param2);
+    });
+
+    test("Signature Help identifies active parameter if there is no comma", async function () {
```
Comment (rchande) [inline test/integrationTests/signatureHelp.integration.test.ts]: Nice! BTW, where did these tests come from?

Comment (akshita31) [reply test/integrationTests/signatureHelp.integration.test.ts]: I wrote them myself.

**Pull request merged**